shared/src/domain-logic/opportunityFields.test.ts

```diff
@@ -0,0 +1,240 @@
+import { generateMunicipality, generateProfileData } from "../test";
+import { getFieldsForProfile, isEveryFundingFieldAnswered } from "./opportunityFields";
+
+describe("opportunityFields", () => {
+  describe("getFieldsForProfile", () => {
+    it(
+      "returns only date of formation and home based business keys when it is a " +
+        "foreign business with a business type of nexus and a legal structure that is not a trade name and the location is not in NJ",
+      async () => {
+        const foreignNexusBusinessWithoutLocationInNJ = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: false,
+          legalStructureId: "limited-partnership",
+        });
+
+        const profileFields = getFieldsForProfile(foreignNexusBusinessWithoutLocationInNJ);
+        expect(profileFields).toContain("dateOfFormation");
+        expect(profileFields).toContain("homeBasedBusiness");
+        expect(profileFields).not.toContain("municipality");
+        expect(profileFields).not.toContain("existingEmployees");
+        expect(profileFields).not.toContain("ownershipTypeIds");
+      }
+    );
+
+    it(
+      "returns only date of formation and municipality keys when it is a " +
+        "foreign business with a business type of nexus and a legal structure that is not a trade name and the location is in NJ",
+      async () => {
+        const foreignNexusBusinessWithoutLocationInNJ = generateProfileData({
```

Comment:
[pebble] Copy paste error? This says "without" location in NJ, but it's true

Reply:
changed to foreignNexusBusiness for all of the,